Classes/Service/AssetService.php

```diff
@@ -34,15 +33,22 @@ class AssetService
     /**
      * @param AssetModuleConfigurationDto $configurationDto
      *
-     * @return array<AssetInterface>
+     * @return ResultCollectionDto
      */
-    public function getAssetsThatNeedProcessing(AssetModuleConfigurationDto $configurationDto): array
+    public function getAssetsThatNeedProcessing(AssetModuleConfigurationDto $configurationDto): ResultCollectionDto
```

Comment:
Ich les gerade nur den PR online, ich würde erwarten dann man eine AssetModuleConfigurationDto reingibt und eine AssetModuleResultCollectionDto raus kriegt.. oder wird die auch für andere Configurations genutzt? Dann AssetResultCollectionDto?

Reply:
Das ist generisch, weil die `ResultCollectionDto` nur zwei Properties hat: `items` und `nextFirstResult`. Ich sah hier keinen Grund das noch einmal weiter zu unterteilen. Falls du meinst es macht mehr Sinn das expliziter zu machen, kann ich das gerne noch ändern. 

Eventuell könnte aber `ResultCollectionDto<AssetModuleResultDto>` gehen. Ich muss mal schauen, wie das die Doctrine Collections lösen.

Reply:
Ja, jetzt, wo ich den Frontend-Code besser kenne, macht das so eh Sinn. 

Das Naming "getAssetsThatNeedProcessing" is etwas seltsam. Das ist einfach eine Suche basierend auf Filtern. Vielleicht einfach:
```php
public function find(AssetFilterDto $filters): ResultCollectionDto
```